native/src/routes/news/components/NewsDetail.js

```diff
@@ -43,11 +44,23 @@ const NewsHeadLine: StyledComponent<{||}, ThemeType, *> = styled.Text`
 type PropsType = {|
   theme: ThemeType,
   language: string,
-  selectedNewsItem: TunewsModel | LocalNewsModel
+  newsItem: TunewsModel | LocalNewsModel,
+  navigateToLink: (url: string, language: string, shareUrl: string) => void
 |}
 
-const NewsDetail = ({ theme, selectedNewsItem, language }: PropsType) => {
-  const content = selectedNewsItem.content || selectedNewsItem.message || ''
+const NewsDetail = ({ theme, newsItem, language, navigateToLink }: PropsType) => {
+  const localNewsContent = newsItem instanceof LocalNewsModel ? newsItem.message : ''
+  const tuNewsContent = newsItem instanceof TunewsModel ? newsItem.content : ''
+  const content = localNewsContent || tuNewsContent
+
+  // https://stackoverflow.com/a/3809435
+  const regex = /((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w\-_]*)?\??(?:[-+=&;%@.\w_]*)#?(?:[.!/\\\w]*))?)/g
```

Comment:
Move this to a constant variable for performance reasons. That way the RegExp object gets created only once.

Reply:
:+1: Moved the whole thing to the api-client. Better for testing, modularity and reusability anyway.